### ISSUE TITLE
[Button] Focusing buttons and menu-buttons in Safari doesn't work by default without accessibility option selected

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#3186]
### Provide a general summary of the issue here

Focusing buttons and menu-buttons with keyboard is not possible in Safari without having accessibility settings set:

<img width="823" height="557" alt="Image" src="https://github.com/user-attachments/assets/ed6b9a60-bda8-49fd-914c-046c6cd3a0d1" />

Focusing a button or menu-button in the tool-bar works fine, also focusing toggle button works ok.

https://github.com/user-attachments/assets/2185bb37-f500-4f32-a18d-9fe8c03775b6

### 👤 Your name

SLDS Team

### 🧢 Your Product/Team

SLDS

**Comment by a11ymiko** [comment]
Tests on Safari with "Press Tab to highlight..." option turned on showed that user can use keyboard to navigate to components: 

1. Button
2. Menu button
3. Button bar
4. Dialog
5. Toolbar
6. Tooltip

That components were pointed as changed with #3139 Add support for the Invoker Commands API in button and dialog PR

`sl-button` and `buttons` don't have unnecessary `tabindex=0`. This attribute is only used for `button` in Toolbar where it's necessary.

**Comment by Diaan** [comment]
We will add a small not about this particular safari behaviour on https://sanomalearning.design/categories/guidelines/accessibility/

Instead of the notes about the dialog focus trap, since that is no longer true.
